modules/light-clients/07-tendermint/store.go

```diff
@@ -80,7 +80,6 @@ func IterateConsensusMetadata(store sdk.KVStore, cb func(key, val []byte) bool)
 	iterator := sdk.KVStorePrefixIterator(store, []byte(host.KeyConsensusStatePrefix))
 
 	// iterate over processed time and processed height
-	defer iterator.Close()
```

Comment:
I think the defer pattern is still better to use, I worry that calling iterator.Close() could later be accidentally re-converted to a defer statement as it is typically more idiomatic. We can make defer version work in a few ways, either create a new variable to defer 

e.g.
```golang
iter := iterator
defer iter.Close()
```

or use a new variable name below rather than re-assign. (I don't see any reason why we would need to re-assign below.)

WDYT?

Reply:
Nice suggestion, I agree sticking to the defer pattern is probably best. I've pushed an update using a new variable instead of reassignment.